crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -542,10 +603,20 @@ fn check_device_status_aws(
                 acknowledged = true;
             }
             Ok(Event::Incoming(Packet::Publish(response))) => {
-                // We got a response
-                println!("Received expected response on topic {}.", response.topic);
-                exists = true;
-                break;
+                if response.topic == aws_topic_sub_check_connection {
+                    // We got a response
+                    println!("Received expected response on topic {}.", response.topic);
+                    exists = true;
+                    break;
+                } else if is_bridge_health_up_message(&response, &built_in_bridge_health) {
+                    // Built in bridge is now up, republish the message in case it was never received by the bridge
+                    client.publish(
+                        &aws_topic_pub_check_connection,
+                        AtLeastOnce,
+                        false,
+                        REGISTRATION_PAYLOAD,
+                    )?;
+                }
```

Comment:
Isn't this logic valid for the mosquitto bridge as well? Also wondering why don't we move the `is_bridge_health_up_message` check before entering this event loop or even before the test client connection is established.

Reply:
For the mosquitto bridge, I don't think this is true. We don't need the cloud connection to exist; we just need the bridge to have a subscription to the message. I haven't read into this to verify it, but my understanding is that the mosquitto bridge is subscribed to the relevant topics before mosquitto is up for others to listen to.

In the same vein, we arguably don't need to check that the built-in bridge health is up, a down message, even a retained one, proves that the subscription exists (even if it's a persisted retained message, as it's not a clean session). That said, using that as a check would be both confusing and not gain us much since we need the cloud connection anyway to make progress.